pkg/clients/ec2/routetable.go

```diff
@@ -143,46 +158,68 @@ func LateInitializeRT(in *v1alpha4.RouteTableParameters, rt *ec2.RouteTable) { /
 	}
 }
 
-// CreateRTPatch creates a *v1alpha4.RouteTableParameters that has only the changed
-// values between the target *v1alpha4.RouteTableParameters and the current
+// CreateRTPatch creates a *v1alpha5.RouteTableParameters that has only the changed
+// values between the target *v1alpha5.RouteTableParameters and the current
 // *ec2.RouteTable
-func CreateRTPatch(in ec2.RouteTable, target v1alpha4.RouteTableParameters) (*v1alpha4.RouteTableParameters, error) {
-	currentParams := &v1alpha4.RouteTableParameters{}
+func CreateRTPatch(in ec2.RouteTable, target v1alpha5.RouteTableParameters) (*v1alpha5.RouteTableParameters, error) {
+	targetCopy := target.DeepCopy()
+	currentParams := &v1alpha5.RouteTableParameters{}
 
 	v1beta1.SortTags(target.Tags, in.Tags)
 
 	// Add the default route for fair comparison.
 	for _, val := range in.Routes {
-		if *val.GatewayId == LocalGatewayID {
-			target.Routes = append([]v1alpha4.Route{{
+		if val.GatewayId != nil && *val.GatewayId == LocalGatewayID {
+			target.Routes = append([]v1alpha5.Route{{
 				GatewayID:            val.GatewayId,
 				DestinationCIDRBlock: val.DestinationCidrBlock,
 			}}, target.Routes...)
 		}
 	}
+	SortRoutes(targetCopy.Routes, in.Routes)
```

Comment:
nitpick: you can sort the routes in `cmp.Equal` by using `cmpopts.SortSlices()` option instead of sorting them separately.

Reply:
Was using this to make sure they were sorted before creating a patch instead of at the compare bit (if that makes sense) was seeing issues on the patch if they weren't in the same order.